src/Explorer/Explorer.tsx

```diff
@@ -282,6 +282,42 @@ export default class Explorer {
     }
   }
 
+  public openInVsCode(): void {
+    const activeTab = useTabs.getState().activeTab;
+    const resourceId = encodeURIComponent(userContext.databaseAccount.id);
+    const database = encodeURIComponent(activeTab?.collection?.databaseId);
+    const container = encodeURIComponent(activeTab?.collection?.id());
+    const downloadUrl = "https://code.visualstudio.com/download";
+
+    const baseUrl = `vscod://ms-azuretools.vscode-cosmosdb?resourceId=${resourceId}`;
+    const vscodeUrl = activeTab ? `${baseUrl}&database=${database}&container=${container}` : baseUrl;
+    const vscodeInsidersUrl = vscodeUrl.replace("vscode://", "vscode-insiders://");
+
+    const startTime = Date.now();
+    let hasRedirected = false;
+
+    const iframe = document.createElement("iframe");
```

Comment:
Even with an Iframe I think this is too error prone and browser behaviour is not really predictable. I'd recommend the following:
* Open a simple Dialog with following content:
   * Title "Opening Visual Studio Code"
   * Message "If nothing happens [install Visual Studio Code]("https://code.visualstudio.com/download") and try again.
   * Buttons: "Cancel" "Download Visual Studio Code"
* When Dialog opens, redirect to the VS Code URI (use current window, don't taget blank)
* If navigation succeeds, just close the dialog. If VS Code doesn't open, user can decide to go and download or cancel.

Reply:
with your latest fix it ([6c5e7f2](https://github.com/Azure/cosmos-explorer/pull/2116/commits/6c5e7f21d7002d31e672fd983aa0d384c7e07d67)) still doesn't work quite right unfortunately: [see above](https://github.com/Azure/cosmos-explorer/pull/2116#discussion_r2064321069)

Reply:
Hhhm, also with the increased timeout (32adf7fde66030c69fc440a3dece0c0b6b7c6701) no luck, now though the download page never opens (tried uninstalling VS Code, then even nothing happens when I click the button) and the dialog is shown twice. I looked into how GitHub is doing it, and they basically show a div that gets removed if opening succeeded. TBH, let's go with the dialog, that'd be the simplest.